src/jvmMain/kotlin/com/charleskorn/kaml/YamlOutput.kt

```diff
@@ -180,6 +181,21 @@ internal class YamlOutput(
         return typeName
     }
 
+    private fun String.isAmbiguous(): Boolean {
+        return when {
+            isEmpty() -> true
+            toBigIntegerOrNull() != null -> true
+            startsWith("0x") && removePrefix("0x").toBigIntegerOrNull(16) != null -> true
+            startsWith("0o") && removePrefix("0o").toBigIntegerOrNull(8) != null -> true
```

Comment:
Anything that starts with `0x` or `0o` is ambiguous, even if the remainder is not a valid integer.